### ISSUE TITLE
Disable warning 'no Await in Async function"

### ISSUE THREAD

**Issue by vbcodec** [dotnet/roslyn#8163]
I have base class with few virtual async functions, and derived classes where overrided function mostly do not use await. All of this creates long list with this warning. Think that no such warning should exist, because there is no way that something can go wrong - producing error. Eventually there should be third category - suggestions. You may change meaning of this warning, by movie it to suggestions, and display only if async function is called without Await inside some other async functions.

**Comment by TKharaishvili** [comment]
@vbcodec I believe, you can override the virtual functions without using the async modifier in the overriding functions, just matching the method signature is sufficient.

**Comment by AdamSpeight2008** [comment]
You could disable the warning.
```vb
#Disable Warning BC42356 ' This async method lacks 'Await' operators and so will run synchronously
    Overridable Async Function Foo() As Task(Of Object)
#Enable Warning BC42356 ' This async method lacks 'Await' operators and so will run synchronously
```

**Comment by srivatsn** [comment]
You could also use the ruleset editor to change the warning level of this warning to an Info so that it just shows up in the messages tab of the error list (similar to your "suggestions" suggestion).
